Classes/Notifications/NotificationsViewController.swift

```diff
@@ -63,6 +63,18 @@ NotificationNextPageSectionControllerDelegate {
             target: self,
             action: #selector(NotificationsViewController.onMarkAll(sender:))
         )
+        updateMarkAllEnabled()
+    }
+    
+    private func updateMarkAllEnabled() {
+        var allRead = true
+        for notification in filteredNotifications {
```

Comment:
Maybe a bit simpler to just do a `.contains` on `filteredNotifications`. Example:

```
let allRead = !filteredNotifications.contains(where: { $0.read == false })
```

(Not sure if that logic is the right way round, but for example it'll do!